pkg/build/api/v1/types.go

```diff
@@ -402,6 +408,19 @@ type SourceBuildStrategy struct {
 	ForcePull bool `json:"forcePull,omitempty"`
 }
 
+// JenkinsPipelineBuildStrategy holds parameters specific to a Jenkins Pipeline build.
+type JenkinsPipelineBuildStrategy struct {
+	// JenkinsfilePath is the path of the Jenkinsfile that will be used to configure the pipeline
+	// relative to the root of the source repository.
+	JenkinsfilePath string `json:"jenkinsfilePath,omitempty"`
+
+	// Jenkinsfile is the raw contents of a Jenkinsfile containing the pipeline definition.
+	Jenkinsfile string `json:"jenkinsfile,omitempty"`
+
+	// Env contains additional environment variables you want to pass to the Jenkins Pipeline.
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
Yeah - I was hoping we'd put the Environment metadata: which cluster URL, which namespace/project each logical environment maps to (UAT / Staging / Production etc) and whether the development environment can communicate direct with the environment or if it needs to put the kubernetes resources into a git repository (directly or via Pull Request) - into environment variables so that the Jenkinsfile can be agnostic of that stuff.

Then we can have reusable pipelines that grok the building/testing stuff and reuses configuration from the BuildConfig to know how to deploy to each environment. 

I guess the other option could be a specific DSL step 'deploy' which talks to the BuildConfig to get the data...